src/main/java/com/google/api/generator/spring/composer/SpringPropertiesClassComposer.java

```diff
@@ -75,7 +75,18 @@ public GapicClass generate(GapicContext context, Service service) {
     String className = Utils.getServicePropertiesClassName(service);
     GapicServiceConfig gapicServiceConfig = context.serviceConfig();
     Map<String, TypeNode> dynamicTypes = createDynamicTypes(service, packageName);
-    boolean hasRestOption = context.transport().equals(Transport.GRPC_REST);
```

Comment:
How is this property used in this composer? Why we don't need this logic in https://github.com/googleapis/gapic-generator-java/pull/1343?

Reply:
@blakeli0 The changes from #1343 removed the generation of SpringAutoConfiguration code that condition on this property (`if (this.clientProperties.getUseRest())`) to set rest-specific defaults. (e.g. [this change to TetherSpringAutoconfiguration](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1586/files#diff-332aa1300b6692a2a98b02f50e233405f3a483a752f68f02878f5a2fb5e41119))

The [property](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-previews/google-cloud-apigee-connect-spring-starter/src/main/java/com/google/cloud/apigeeconnect/v1/spring/TetherSpringProperties.java#L41-L42) itself is still there (just ignored by the autoconfig), and will be removed as part of this PR's changes. I realized when updating these tests that we shouldn't be generating the property either, and it was something I had overlooked in #1343's patch.

Reply:
I see, thanks for the explanation! It was an unused property hence not causing any compilation issues. 

I did a quick search for `rest` and see that it is mentioned in this [comment](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/c215b9fa8105920fccf09d3be78fc79d195d6089/spring-cloud-previews/google-cloud-apigee-connect-spring-starter/src/main/java/com/google/cloud/apigeeconnect/v1/spring/TetherSpringAutoConfiguration.java#L88) , we may want to update it to something like `unless the useRest option is supported and provided`, as not every client library support rest.

Reply:
That makes sense - I've updated the comment composer correspondingly. Thanks for the suggestion!